projects/noloan2/app/src/main/java/hasadna/noloan2/MainActivity.java

```diff
@@ -67,11 +80,44 @@ private FirestoreElement encodeUser(loguser user) {
   @Override
   public void onClick(View v) {
     switch (v.getId()) {
-      case R.id.B_adduser:
+      case R.id.B_adduser: // Clicking on the add user button
         loguser user = loguser.newBuilder()
           .setName(name.getText().toString())
           .build();
         writeUser(user);
     }
   }
+  
+  // The user can change the permission after the splash screen, so checking for permission is still needed.
+  private ArrayList<SMSmessage> readSms() {
+    // Check for permission reading sms
+    int permissionStatus = checkSelfPermission(Manifest.permission.READ_SMS);
+    if (permissionStatus != PackageManager.PERMISSION_GRANTED) // If don't have permission show toast and close the app
```

Comment:
Try to stay within 100 characters per line.
If it's a long comment, you can put it before the `if`.